src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+import { User } from '../../types/User';
+
+const DEFAULT_INPUTS = {
+  title: '',
+  userId: 0,
+};
+
+const DEFAULT_INPUTS_ERRORS = {
+  title: false,
+  userId: false,
+};
+
+interface Props {
+  onAdd: (newTodoInfo: {
+    title: string,
+    userId: number,
+  }) => void;
+  users: User[],
+}
+
+export const TodoForm: React.FC<Props> = ({
+  onAdd = () => { },
+  users,
+}) => {
+  const [inputs, setInputs] = useState(DEFAULT_INPUTS);
+  const [inputsErrors, setInputsErrors] = useState(DEFAULT_INPUTS_ERRORS);
+  const handleInputChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+    | React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    const { name, value } = event.target;
+
+    const newValue = name === 'userId' ? +value : value;
+
+    setInputs(prevFields => ({
+      ...prevFields,
+      [name]: newValue,
+    }));
+
+    setInputsErrors(prevErrors => ({
+      ...prevErrors,
+      [name]: !newValue,
+    }));
+  };
+
+  const resetForm = () => {
+    setInputs(DEFAULT_INPUTS);
+    setInputsErrors(DEFAULT_INPUTS_ERRORS);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setInputsErrors(() => ({
+      title: !inputs.title.trim(),
+      userId: !inputs.userId,
+    }));
+
+    setInputs(prevInputs => ({
+      ...prevInputs,
+      title: prevInputs.title.trim(),
+    }));
+
+    if (!inputs.title.trim() || !inputs.userId) {
+      return;
+    }
+
+    onAdd({
+      ...inputs,
+    });
```

Comment:
```suggestion
    onAdd(inputs);
```
You can just pass the object directly